tools/verify-release/verify-release.sh

```diff
@@ -0,0 +1,534 @@
+failures_file="$(pwd)/${run_id}.log"
+
+dist_url_prefix="https://dist.apache.org/repos/dist/dev/incubator/polaris/"
+keys_file_url="https://downloads.apache.org/incubator/polaris/KEYS"
+
+version_full="${version}-incubating"
+git_tag_full="apache-polaris-${version_full}-rc${rc_num}"
+
+GITHUB=0
+[[ -n ${GITHUB_ENV} ]] && GITHUB=1
+
+# Common excludes for "find'
+find_excludes=(
+  # Exclude GPG signatures and checksums
+  '!' '-name' '*.asc'
+  '!' '-name' '*.md5'
+  '!' '-name' '*.sha1'
+  '!' '-name' '*.sha256'
+  '!' '-name' '*.sha512'
+  # file with that name is created by wget when mirroring from 'dist'
+  '!' '-name' "${version_full}"
+  # ignore Maven repository metadata
+  '!' '-name' 'maven-metadata*.xml'
+  '!' '-name' 'archetype-catalog.xml'
+)
+
+dist_url="${dist_url_prefix}${version_full}"
+helm_url="${dist_url_prefix}helm-chart/${version_full}"
+maven_repo_url="${maven_repo_url_prefix}${maven_repo_id}/"
+
+function log_part_start {
+  local heading
+  local separator
+  heading="${*}"
+  [ ${GITHUB} == 1 ] && echo "::group::$heading"
+  echo ""
+  # shellcheck disable=SC2046
+  separator="--$(printf -- '-%0.s' $(eval "echo {1..${#heading}}"))--"
+  echo "${separator}"
+  echo "  ${heading}"
+  echo "${separator}"
+}
+
+function log_part_end {
+  [ ${GITHUB} == 1 ] && echo "::endgroup::"
+  echo ""
+}
+
+function log_fatal {
+  echo -n -e "${RED}"
+  echo -n "$1"
+  echo -e "${RESET}"
+  echo "" >> "${failures_file}"
+  for i in "${@}"; do
+    echo "$i" >> "${failures_file}"
+  done
+}
+
+function log_warn {
+  echo -ne "${ORANGE}"
+  echo -n "$1"
+  echo -e "${RESET}"
+}
+
+function log_info {
+  echo "$1"
+}
+
+# Executes a process and captures a fatal error if the process did not complete successfully.
+# The full output of the process execution will be logged in the FAILURES file, but not printed to the console.
+#   First argument: log message for 'log_fatal'
+#   Following arguments: process arguments
+function proc_exec {
+  local err_msg
+  local output
+  err_msg=$1
+  shift
+  output=()
+  IFS=$'\n' read -r -d '' -a output < <( "${@}" 2>&1 && printf '\0' ) || (
+    log_fatal "${err_msg}" "${output[@]}"
+    return 1
+  )
+}
+
+function mirror {
+  local url
+  local dir
+  local cut
+  local wget_executable
+  url="$1"
+  dir="$2"
+  cut="$3"
+  wget_executable="wget"
+  # Prefer wget2 as it allows parallel downloads (wget does not)
+  (which wget2 > /dev/null) && wget_executable="wget2 --max-threads=8"
+  log_part_start "Mirroring $url, this may take a while..."
+  mkdir -p "${dir}"
+  (cd "${dir}" ; ${wget_executable} \
+    --no-parent \
+    --no-verbose \
+    --no-host-directories \
+    --mirror \
+    -e robots=off \
+    --cut-dirs="${cut}" \
+    "${url}/")
+  # Nuke the directory listings (index.html from server) and robots.txt...
+  # (only wget2 downloads the robots.txt :( )
+  find "${dir}" \( -name index.html -o -name robots.txt \) -exec rm {} +
+  find "${dir}" -name "*.prov" | while read -r helmProv; do
```

Comment:
Could you clarify why this is necessary?  `helmProv` makes me think that you want the helm chart related `.prov` files.  But there are other such files in a release.  And AFAIK they are not gzipped.